map/lanelet2_extension/test/src/test_route_checker.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lanelet2_extension/utility/message_conversion.hpp"
+#include "lanelet2_extension/utility/route_checker.hpp"
+
+#include <autoware_auto_mapping_msgs/msg/had_map_bin.hpp>
+#include <autoware_auto_planning_msgs/msg/had_map_route.hpp>
+
+#include <gtest/gtest.h>
+
+using lanelet::Lanelet;
+using lanelet::LineString3d;
+using lanelet::Point3d;
+using lanelet::utils::getId;
+
+class TestSuite : public ::testing::Test
+{
+public:
+  TestSuite() : sample_map_ptr(new lanelet::LaneletMap())
+  {
+    // NOLINT
+    // create sample lanelets
+    Point3d p1, p2, p3, p4;
+
+    p1 = Point3d(getId(), 0., 0., 0.);
+    p2 = Point3d(getId(), 0., 1., 0.);
+
+    LineString3d ls_left(getId(), {p1, p2});  // NOLINT
+
+    p3 = Point3d(getId(), 1., 0., 0.);
+    p4 = Point3d(getId(), 1., 1., 0.);
```

Comment:
[IMO] This style is widely recommended.

```suggestion
    const Point3d p1(getId(), 0.0, 0.0, 0.0);
    const Point3d p2(getId(), 0.0, 1.0, 0.0);

    const LineString3d ls_left(getId(), {p1, p2});

    const Point3d p3(getId(), 1.0, 0.0, 0.0);
    const Point3d p4(getId(), 1.0, 0.0, 0.0);
```

Reply:
Thank you! I'll use `const`.